map.go

```diff
@@ -1116,7 +1116,7 @@ func (mi *MapIterator) Next(keyOut, valueOut interface{}) bool {
 		return false
 	}
 
-	for ; mi.count < mi.maxEntries; mi.count++ {
+	for mi.count < mi.maxEntries+1 {
```

Comment:
```suggestion
	for mi.count <= mi.maxEntries {
```

Reads a bit easier personally: 'reading up to `maxEntries`' instead of 'reading up to less than `maxEntries`, plus one'.